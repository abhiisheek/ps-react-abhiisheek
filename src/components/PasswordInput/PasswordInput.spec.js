import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import PasswordInput from './PasswordInput';

describe('PasswordInput', () => {
  it('Should match the snapshot', () => {
    const snapshot = renderer
      .create(
        <PasswordInput
          htmlId="password-input-example-all-features"
          name="password"
          onChange={() => {}}
          value="valureFor#123"
        />
      )
      .toJSON();

    expect(snapshot).toMatchSnapshot();
  });

  it('toggle password visiblity on clicking the show icon', () => {
    const wrapper = shallow(
      <PasswordInput
        htmlId="password-input-example-all-features"
        name="password"
        onChange={() => {}}
        value="valureFor#123"
        showVisibilityToggle
      />
    );

    expect(wrapper.find({ type: 'password' })).toHaveLength(1);
    expect(wrapper.find({ type: 'text' })).toHaveLength(0);

    wrapper.find('a').simulate('click');

    expect(wrapper.find({ type: 'password' })).toHaveLength(0);
    expect(wrapper.find({ type: 'text' })).toHaveLength(1);
  });
});
