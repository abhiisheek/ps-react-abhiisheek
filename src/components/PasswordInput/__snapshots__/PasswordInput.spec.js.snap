// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput Should match the snapshot 1`] = `
<div
  style={
    Object {
      "marginBottom": 16,
    }
  }
>
  <label
    htmlFor="password-input-example-all-features"
    style={
      Object {
        "display": "block",
      }
    }
  >
    Password
     
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      *
    </span>
  </label>
  <input
    id="password-input-example-all-features"
    maxLength={50}
    name="password"
    onChange={[Function]}
    placeholder={undefined}
    style={undefined}
    type="password"
    value="valureFor#123"
  />
</div>
`;
